File: mockito-tips-and-tricks/src/main/java/de/rieckpil/blog/staticsmocks/OrderService.java
package de.rieckpil.blog.staticsmocks;

import java.time.LocalDateTime;
import java.util.UUID;

public class OrderService {

  public Order createOrder(String productName, Long amount, String parentOrderId) {
    Order order = new Order();

    order.setId(parentOrderId == null ? UUID.randomUUID().toString() : parentOrderId);
    order.setCreationDate(LocalDateTime.now());
    order.setAmount(amount);
    order.setProductName(productName);

    return order;
  }
}

File: mockito-tips-and-tricks/src/main/kotlin/de/rieckpil/blog/kt/OrderService.kt
package de.rieckpil.blog.kt

import java.time.LocalDateTime
import java.util.*

class OrderService {

  fun createOrder(productName: String, amount: Long, parentOrderId: String?) =
    Order(
      id = parentOrderId ?: UUID.randomUUID().toString(),
      creationDateTime = LocalDateTime.now(),
      amount = amount,
      productName = productName
    )
}

data class Order(
  val productName: String,
  val amount: Long,
  val id: String,
  val creationDateTime: LocalDateTime
)
